src/Components/Products/index.jsx

```diff
@@ -1,36 +1,51 @@
-import { useSelector } from 'react-redux';
-import { Card, CardActions, CardContent, CardMedia, Button, Typography } from '@mui/material';
+import { useSelector, useDispatch } from 'react-redux';
+import { Card, CardActions, CardContent, CardMedia, Button, Grid, Typography } from '@mui/material';
+import { When } from 'react-if';
+import { addToCart } from '../../store/action';
+import '../../../App.css';
 
 function Products() {
     const { products } = useSelector((state) => state.products);
-    console.log('these are the products', products);
+    const { activeCategory } = useSelector((state) => state.categories);
+    const dispatch = useDispatch();
 
     return (
-        products.map((product, index) => (
-            <>
-            {
-                <Card key={`products-${index}`} sx={{ maxWidth: 345 }}>
-                    <CardMedia id="img-container"
-                    sx={{ height: 140 }}
-                    image={`https://source.unsplash.com/random/?${product.name}}`}
-                    title="random image"
-                    />
-                    <CardContent>
-                        <Typography gutterBottom variant="h5" component="div">
-                            Price: {product.price}
-                        </Typography>
-                        <Typography variant="body2" color="text.secondary">
-                            In-Stock: {product.inStock}
-                        </Typography>
-                    </CardContent>
-                    <CardActions>
-                        <Button size="small">Add to Cart</Button>
-                        <Button size="small">View Details</Button>
-                    </CardActions>
-                </Card>
-            }
-            </>
-        ))
+        <When condition={activeCategory}>
+            <Grid container spacing={2} width="80%" margin="auto">
+                {products.map((product, index) => (
+                    <>
+                        {
+                            <Grid item xs={12} sm={6} md={6} lg={4}>
+                                <Card key={`products-${index}`} sx={{ maxWidth: 345 }}>
```

Comment:
Your key property needs to be in the outermost element of you map. The fragment and curly braces are unnecessary, so I would get rid of them. You can then either move the key prop to the grid element or ditch the grid element as well.

Reply:
Are you talking about line 16 and 17? Or just on line 19? Cause deleting them on 19 breaks everything.

Reply:
Yes, line 16 and 17 (and their closers) can be deleted then move the key prop.